examples/prediction/xgboost-high-level-apis.ipynb

```diff
@@ -0,0 +1,307 @@
+   "metadata": {},
+   "source": [
+    "## Training Locally"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "HousingServe().train()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Fairing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import fairing\n",
+    "\n",
+    "# Setting up google container repositories (GCR) for storing output containers\n",
+    "# You can use any docker container registry istead of GCR\n",
+    "GCP_PROJECT = fairing.cloud.gcp.guess_project_name()\n",
+    "DOCKER_REGISTRY = 'gcr.io/{}/fairing-job'.format(GCP_PROJECT)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Build base image for remote training and prediction"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "py_version = \".\".join([str(x) for x in sys.version_info[0:3]])\n",
+    "base_image = 'gcr.io/{}/fairing-predict-example:latest'.format(GCP_PROJECT)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!docker build --build-arg PY_VERSION={py_version} . -t {base_image}\n",
+    "!docker push {base_image}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training in KF"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "from fairing.notebook.jobs import TrainJob, guess_docker_registry\n",
+    "train_job = TrainJob(HousingServe, base_image, input_files=['ames_dataset/train.csv'])\n",
+    "train_job.submit()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Deploying model and creating an endpoint in KF"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "from fairing.notebook.jobs import PredictionEndpoint\n",
+    "endpoint = PredictionEndpoint(HousingServe, base_image, input_files=['trained_ames_model.dat'])\n",
+    "endpoint.create()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Making prediction calls against the endpoint"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Copy the prediction endpoint from prev step\n",
+    "!curl http://146.148.70.8:5000/predict -H \"Content-Type: application/x-www-form-urlencoded\" -d 'json={\"data\":{\"tensor\":{\"shape\":[1,37],\"values\":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37]}}}'\n",
```

Comment:
Hoping for this to be pythonic way to make predictions...

Reply:
yes it will be. #124